catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1147,6 +1155,41 @@ public CreateResponse create(CreateRequest createRequest)
         return createResponse;
     }
 
+    private void applyAttributeOverridesToMetacardMap(String attributeOverrideHeaders,
+            Map<String, Metacard> metacardMap) {
+
+        Map<String, String> attributeOverrideMap = createAttributeOverrideMapFromHeaders(
+                attributeOverrideHeaders);
+
+        if (MapUtils.isEmpty(attributeOverrideMap) || MapUtils.isEmpty(metacardMap)) {
+            return;
+        }
+
+        attributeOverrideMap.forEach((attributeMapKey, attributeMapValue) -> metacardMap.forEach((metacardMapKey, metacardMapValue) -> {
+            if (metacardMapValue.getMetacardType()
+                    .getAttributeDescriptor(attributeMapKey) != null) {
+                metacardMapValue.setAttribute(new AttributeImpl(attributeMapKey,
+                        attributeMapValue));
+            }
+        }));
+    }
+
+    private Map<String, String> createAttributeOverrideMapFromHeaders(
```

Comment:
I don't think the catalogframework should be the one to build this map - it should get a map from the properties.